### PR TITLE
Reorganize controls, export react controllers

### PR DESCRIPTION
**Pull request by ibgreen**
Another preparatory PR

### CODE REVIEW

File: examples/without-map/app.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /* global window,document,fetch */
 import React, {Component} from 'react';
 import {render} from 'react-dom';
-import {MapController} from 'deck.gl/dist/controllers';
+import {MapController} from 'deck.gl';
```
Comment (Pessimistress) [inline examples/without-map/app.js]: Should we be updating the examples at this stage? This app will not work with `npm start`. 
We also discussed about not depending on alpha packages in examples. @shaojingli

File: src/react/controllers/map-controller.js [hunk 2]
```diff
@@ -60,7 +60,7 @@ const propTypes = {
   // A map control instance to replace the default map controls
   // The object must expose one property: `events` as an array of subscribed
   // event names; and two methods: `setState(state)` and `handle(event)`
-  mapControls: PropTypes.shape({
+  controls: PropTypes.shape({
```
Comment (Pessimistress) [inline src/react/controllers/map-controller.js]: Are you proposing we change the interface in react-map-gl as well?

File: examples/without-map/app.js (hunk 1 above)
Comment (ibgreen) [inline examples/without-map/app.js]: > Should we be updating the examples at this stage? This app will not work with npm start.

True. At the same time, importing from dist is very problematic in a multi-distribution package like deck.gl. Depending on which bundler is used, one can easily get the main library and these files from different trees (e.g. dist vs. dist-es6), which can lead to very subtle and unpleasant bugs.

> We also discussed about not depending on alpha packages in examples. @shaojingli

Hmm, I see. I understand the thinking, but having to hold off changing examples until we release is a little tricky. 

I really want to isolate the react wrappers into the react directory. I guess I could add one line forwarding export files in the old directory to make the old imports work.

File: src/react/controllers/map-controller.js (hunk 2 above)
Comment (ibgreen) [inline src/react/controllers/map-controller.js]: > Are you proposing we change the interface in react-map-gl as well?

No strong opinions on that yet. Just saw an opportunity to make the different controllers more alike.

**Pull request merged**